query causes panic in FieldIDByName

After writing a large volume of synthetic data, doing a `COUNT(value) FROM ...` query will cause the following panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x8 pc=0x168c16]

goroutine 30058 [running]:
github.com/influxdb/influxdb/tsdb.(*FieldCodec).FieldIDByName(0x0, 0xc20bf9cf2a, 0x5, 0x5, 0x0, 0x0)
    /Users/todd/go/src/github.com/influxdb/influxdb/tsdb/shard.go:585 +0x46
github.com/influxdb/influxdb/tsdb.(*LocalMapper).Begin(0xc20844a9a0, 0xc20bb976e0, 0x0, 0x1, 0x0, 0x0)
    /Users/todd/go/src/github.com/influxdb/influxdb/tsdb/tx.go:315 +0x2c4
github.com/influxdb/influxdb/influxql.(*MapReduceJob).processAggregate(0xc20f69cb60, 0xc20bb976e0, 0x769af8, 0xc209b6b860, 0x1, 0x1, 0x0, 0x0)
    /Users/todd/go/src/github.com/influxdb/influxdb/influxql/engine.go:769 +0x14b
github.com/influxdb/influxdb/influxql.(*MapReduceJob).Execute(0xc20f69cb60, 0xc20c2329c0, 0xc208039c00)
    /Users/todd/go/src/github.com/influxdb/influxdb/influxql/engine.go:170 +0x8ca
github.com/influxdb/influxdb/influxql.(*Executor).execute(0xc20cf69200, 0xc20c2329c0)
    /Users/todd/go/src/github.com/influxdb/influxdb/influxql/engine.go:926 +0xba
created by github.com/influxdb/influxdb/influxql.(*Executor).Execute
    /Users/todd/go/src/github.com/influxdb/influxdb/influxql/engine.go:905 +0x5a
```

It's not reliably reproducible, but both @dgnorton and I had seen it earlier this week. I've been able to produce it several times with current master. Starting with fresh data doesn't change anything.

Steps to reproduce:
1. Write data to a single measurement with some tags and a single field named `value` - usually batches of 1k or 10k for 100-1000 iterations.
2. Run a `SELECT COUNT(value) FROM ...` after the data is written - this is the point where the server would panic.
3. Restarting the server in between write batches may be part of the issue.

--- a/tsdb/shard.go
+++ b/tsdb/shard.go
@@ -136,7 +136,10 @@ func (s *Shard) WritePoints(points []Point) error {
 		// marshal the raw data if it hasn't been marshaled already
 		if p.Data() == nil {
 			// this was populated earlier, don't need to validate that it's there.
-			data, err := s.measurementFields[p.Name()].codec.EncodeFields(p.Fields())
+			s.mu.RLock()
+			mf := s.measurementFields[p.Name()]
+			s.mu.RUnlock()
+			data, err := mf.codec.EncodeFields(p.Fields())
 			if err != nil {
 				return err
 			}

